src/librustc_session/lint/builtin.rs

```diff
@@ -46,6 +46,18 @@ declare_lint! {
     "shift exceeds the type's number of bits"
 }
 
+declare_lint! {
+    pub OVERFLOW,
+    Deny,
```

Comment:
I'd lean towards warn-by-default, but `EXCEEDING_BITSHIFTS` was deny-by-default so I stuck with that.

Reply:
I think a refactoring PR isn't the place to change the severity of the lint.

Reply:
This is not a refactoring PR. This is the part that I did not do in https://github.com/rust-lang/rust/pull/68969 because *that* was a refactoring PR.
For example, this PR already moves a bunch of warnings from `const_err` to other lints.

Reply:
That said, I am fine with keeping the severity as-is. I don't have a strong opinion. This just feels more like "warn" material to me than "deny". If @rust-lang/wg-diagnostics agrees, someone can file a follow-up to change the default.